README.md

```diff
@@ -5,17 +5,58 @@ If you don't already have it, install
 
 Before doing anything else, grab your dependencies:
 
-    $ dep ensure
+    $ dep ensure -v
 
-## Create a release
+## Create a Release
 
-Update [the resource files](deploy/resources/) with the proper
-[quay.io](https://quay.io/organization/openshift-knative) images,
-verify that [version.go](version/version.go) is correct, and then run
-these commands:
+Verify that [version.go](version/version.go) matches the contents of
+[deploy/resources](deploy/resources/) and then run the following to
+build and push an image for the operator to
+[quay.io](https://quay.io/repository/openshift-knative/knative-eventing-operator).
 
-    $ VERSION="vX.Y.Z"      # ensure this matches version/version.go!
-    $ operator-sdk build quay.io/openshift-knative/knative-eventing-operator:$VERSION
-    $ docker push quay.io/openshift-knative/knative-eventing-operator:$VERSION
-    $ git tag $VERSION
-    $ git push --tags
+    ./hack/release.sh
+
+## Create a CatalogSource for [OLM](https://github.com/operator-framework/operator-lifecycle-manager)
+
+The OLM requires special manifests that the operator-sdk can help
+generate. First we extract the CRD's from the upstream manifest[s]
+contained within [deploy/resources](deploy/resources/):
+
+    ./hack/extract-crds.sh
+
+Update [deploy/role.yaml](deploy/role.yaml/) to match the RBAC
+policies in the upstream manifest[s].
+
+And then generate a basic `ClusterServiceVersion` passing in a version
+that corresponds to those manifests:
+
+    operator-sdk olm-catalog gen-csv --csv-version $VERSION
+
+Some post-editing of the file it generates is required:
+
+* Add fields to address any warnings it reports
+* Add `description` and `displayName` fields for all owned CRD's
```

Comment:
@jcrossley3 ideally `description` and `displayName` fields are present upstream, right ? Or is this OLM/Operator-SDK/Openshift specific ?

Reply:
Yes, OLM-specific, but feel free to copy/paste from upstream.

Reply:
likely does not fly - I can try, however - but I doubt that upstream wants these fields if there is no value for pure k8s